security-group.tf

```diff
@@ -1,25 +1,34 @@
 resource "aws_security_group" "bastion_ssh" {
-  name_prefix = "${var.bastion_name}-ssh-"
+  name_prefix = "${var.bastion_name}-ssh-egress-"
   description = "Allow inbound Bastion SSH and all outbound traffic"
 
-  ingress {
-    from_port   = 22
-    to_port     = 22
-    protocol    = "tcp"
-    cidr_blocks = var.ssh_cidr_blocks
-  }
-
-  egress {
-    from_port   = 0
-    to_port     = 0
-    protocol    = "-1"
-    cidr_blocks = ["0.0.0.0/0"]
-  }
-
   vpc_id = var.vpc_id
+  # Destroy other non-Terraform-managed rules, if the security group needs to be destroyed.
+  revoke_rules_on_delete = true
 
   lifecycle {
     ignore_changes = [tags]
   }
 }
 
+resource "aws_security_group_rule" "bastion_ssh" {
+  # Only add this rule if the list of ssh_cidr_blocks is set.
+  count             = length(var.ssh_cidr_blocks) > 0 ? 1 : 0
```

Comment:
In tf 0.12+, is for_each the correct way to do this, or is count still acceptable? (not a terraform expert, mostly asking)

Reply:
Count is still used in 0.12 - [this 0.12 upgrade page](https://www.terraform.io/upgrade-guides/0-12.html) refers to a very similar syntax to what I've used:
> Another    count   -related change is that Terraform now requires    count to be assigned a numeric value, and will not automatically convert a boolean value to a number in the interests of clarity. If you wish to use a boolean value to activate or deactivate a particular resource, use the conditional operator to show clearly how the boolean value maps to a number value

I think if I were creating separate SG rules for each IP in `ssh_cidr_blocks`, I'd use `foreach` to iterate that list.